packages/library-legacy/src/connection.ts

```diff
@@ -6398,17 +6447,37 @@ export class Connection {
    * @param filters The program account filters to pass into the RPC method
    * @return subscription id
    */
+  onProgramAccountChange(
+    programId: PublicKey,
+    callback: ProgramAccountChangeCallback,
+    config: ProgramAccountSubscriptionConfig,
+  ): ClientSubscriptionId;
+  /** @deprecated Instead, pass in a {@link ProgramAccountSubscriptionConfig} */
+  // eslint-disable-next-line no-dupe-class-members
   onProgramAccountChange(
     programId: PublicKey,
     callback: ProgramAccountChangeCallback,
     commitment?: Commitment,
     filters?: GetProgramAccountsFilter[],
+  ): ClientSubscriptionId;
+  // eslint-disable-next-line no-dupe-class-members
+  onProgramAccountChange(
+    programId: PublicKey,
+    callback: ProgramAccountChangeCallback,
+    commitmentOrConfig?: Commitment | ProgramAccountSubscriptionConfig,
+    maybeFilters?: GetProgramAccountsFilter[],
   ): ClientSubscriptionId {
+    const {commitment, config} =
+      extractCommitmentFromConfig(commitmentOrConfig);
     const args = this._buildArgs(
       [programId.toBase58()],
       commitment || this._commitment || 'finalized', // Apply connection/server default.
       'base64' /* encoding */,
-      filters ? {filters: filters} : undefined /* extra */,
+      config
+        ? config
+        : maybeFilters
+          ? {filters: maybeFilters}
+          : undefined /* extra */,
```

Comment:
According to TypeScript, you either have a config, or you have commitment/filters, but never both.

Reply:
Little gross, but wfm.